src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1610,10 +1614,61 @@ public static int Sum(this IEnumerable<int> source)
             int sum = 0;
             checked
             {
-                foreach (int v in source) sum += v;
+                if (Vector.IsHardwareAccelerated && source.Count() >= SIMD_THRESHOLD)
```

Comment:
I'll remove the source.Count() check.  Good observation.
On Aug 31, 2015 7:37 AM, "Jon Hanna" <notifications@github.com> wrote:

> In src/System.Linq/src/System/Linq/Enumerable.cs
> <https://github.com/dotnet/corefx/pull/3024#discussion_r38307310>:
>
> > @@ -1610,10 +1614,61 @@ public static int Sum(this IEnumerable<int> source)
> >              int sum = 0;
> >              checked
> >              {
> > -                foreach (int v in source) sum += v;
> > +                if (Vector.IsHardwareAccelerated && source.Count() >= SIMD_THRESHOLD)
>
> Potentially the source may not be replayable, and so it would end up with
> Sum throwing an exception or returning 0 because on the subsequent
> attempt there were no elements.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dotnet/corefx/pull/3024/files#r38307310>.
>